apps/wallet/src/engine/runners/solana/history.ts

```diff
@@ -0,0 +1,318 @@
+import type {
+	Connection,
+	ParsedInstruction,
+	ParsedTransaction,
+	ParsedTransactionMeta,
+	ParsedTransactionWithMeta,
+	PublicKey,
+	TokenBalance,
+} from '@solana/web3.js';
+import type { TransactionHistory } from '@walless/core';
+import { Networks } from '@walless/core';
+import { solana } from '@walless/network';
+import type { TokenDocument, TransactionHistoryDocument } from '@walless/store';
+import { selectors } from '@walless/store';
+import { storage } from 'utils/storage';
+
+import { getMetadata } from './metadata';
+import type { ParsedTokenAccountWithAddress } from './types';
+
+const historyLimit = 20;
+
+export const getTransactionsHistory = async (
+	connection: Connection,
+	wallet: PublicKey,
+	accounts: ParsedTokenAccountWithAddress[],
+) => {
+	const splTokenAccount = accounts
+		.filter((account) => account.tokenAmount.decimals > 0)
+		.map((account) => account.publicKey);
+
+	const accountList = [wallet].concat(splTokenAccount);
+	const signatureList = [
+		...new Set(await getTransactionSignatures(connection, accountList)),
+	];
+	const transactions = (
+		await getTransactionsDetail(connection, signatureList)
+	).slice(0, 20);
```

Comment:
you should slice the signatures instead of fetching details first to slice, It will cause a performance problem

Reply:
Because `signatureList` is signatures from several token accounts that is not in order, it is inconvenient to slice the signatures